php/Plugin.php

```diff
@@ -144,14 +154,26 @@ public function version() {
 	/**
 	 * Sync the plugin version with the asset version.
 	 *
+	 * @param string $file Path to the file for which to check modified time.
+	 *
 	 * @return string
 	 */
-	public function asset_version() {
+	public function asset_version( $file = false ) {
 		if ( $this->is_debug() || $this->is_script_debug() ) {
-			return time();
+			return (string) time();
+		}
+
+		$version = $this->version();
+
+		if ( empty( $version ) ) {
```

Comment:
This method is a generic helper to bump the asset version number with every plugin update so maybe we could introduce a new helper for getting a file-specific version string based on `filemtime()`?